webbpsf/utils.py

```diff
@@ -180,6 +180,39 @@ def setup_logging(level='INFO', filename=None):
 """
 
 
+def auto_download_webbpsf_data():
+    import os
+    import tarfile
+    from pathlib import Path
+    from tempfile import TemporaryDirectory
+    from urllib.request import urlretrieve
+
+    # Create a default directory for the data files
+    default_path = Path.home() / "data" / "webbpsf-data"
+    default_path.mkdir(parents=True, exist_ok=True)
+
+    os.environ["WEBBPSF_PATH"] = str(default_path)
+
+    # Download the data files if the directory is empty
+    if not any(default_path.iterdir()):
+        warnings.warn("WebbPSF data files not found in default location, attempting to download them now...")
+
+        with TemporaryDirectory() as tmpdir:
+            # Download the data files to a temporary directory
+            url = "https://stsci.box.com/shared/static/qxpiaxsjwo15ml6m4pkhtk36c9jgj70k.gz"
+            filename = Path(tmpdir) / "webbpsf-data-LATEST.tar.gz"
+            urlretrieve(url, filename)
+
+            # Extract the tarball
+            with tarfile.open(filename, "r:gz") as tar:
+                tar.extractall(default_path.parent, filter="fully_trusted")
```

Comment:
the filter parameter is implemented with python 3.12.  This repo currently supports >=3.10

Reply:
It is in Python 3.10, see https://docs.python.org/3.10/library/tarfile.html#tarfile.TarFile.extractall.

Reply:
sounds good to me then